operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,560 @@
+            measurement: Measurement::Classification(ClassificationMeasurement {
+                measurement: "raw".into(),
+                classes: hashmap!(0 => "Water Bodies".to_string()),
+            }),
+        };
+        let initialized_operator = InitializedXgboostOperator {
+            result_descriptor: out_desc,
+            sources: init_rasters,
+            model_file_path: self.params.model_file_path,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+impl InitializedRasterOperator for InitializedXgboostOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        // given a set of raster bands, if they have different types,
+        // where should they be converted?
+        let vec_of_rqps: Vec<Box<dyn RasterQueryProcessor<RasterType = f64>>> = self
+            .sources
+            .iter()
+            .map(|init_raster| init_raster.query_processor().unwrap().into_f64())
+            .collect();
+
+        Ok(QueryProcessorOut(Box::new(XgboostProcessor::new(
+            vec_of_rqps,
+            self.model_file_path.clone(),
+        ))))
+    }
+}
+
+struct XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    sources: Vec<Q>,
+    model_file: String,
+}
+
+impl<Q, P> XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    pub fn new(sources: Vec<Q>, model_file_path: String) -> Self {
+        Self {
+            sources,
+            model_file: std::fs::read_to_string(model_file_path).unwrap(),
+        }
+    }
+
+    async fn process_tile_async(
+        &self,
+        bands_of_tile: Vec<Result<Tile<P>, crate::error::Error>>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<PixelOut>> {
+        // gather information
+        let reference_tile = bands_of_tile.first().unwrap();
+        let tile = reference_tile.as_ref().unwrap().clone();
+        let n_rows = tile.grid_shape_array()[0];
+        let n_cols = tile.grid_shape_array()[1];
+        let grid_shape = tile.grid_shape();
+        let props = tile.properties;
+        let n_bands = bands_of_tile.len() as i32;
+        let time = tile.time;
+        let tile_position = tile.tile_position;
+        let global_geo_transform = tile.global_geo_transform;
+
+        // extract the actual data from the tiles
+        let rasters: Vec<Vec<f32>> = bands_of_tile
+            .into_iter()
+            .map(|band| {
+                let band_ok = band.unwrap();
+                let mat_tile = band_ok.into_materialized_tile();
+
+                mat_tile
+                    .grid_array
+                    .inner_grid
+                    .data
+                    .into_iter()
+                    .map(num_traits::AsPrimitive::as_)
+                    .collect::<Vec<f32>>()
```

Comment:
you already converted all the source rasters into f64, why do you convert again? Can't you convert them to f32 directly? Then you could use the `data` here as it is

Reply:
I added a new method for the query_processor in order to achieve that. https://github.com/geo-engine/geoengine/pull/639/commits/5927879ce4601e18e31e87a860066f5d6794581d
I hope it's ok like that, or would there be another way?

Reply:
ok but why do you you still convert the data here? They are already f32

Reply:
I might be missing something obvious here, but `data` is a `Vec<P>` and I need that conversion to get stuff over to xgboost, which doesn't know the `Pixel` trait. What other way could I use here?